Taskfile.yml

```diff
@@ -30,7 +30,6 @@ tasks:
       - gofumpt -w .
       - go mod tidy
       - golangci-lint run --fix
-      - nilaway -fix -test=false -include-pkgs='github.com/opslevel/opslevel-go' ./...
```

Comment:
@davidbloss yes - this just slows down CI. `nilaway -fix` is useless in its current state since it's still in alpha.

Reply:
But this is part of `task fix` which is not used in CI. If `nilaway -fix` isn't useful today but becomes useful later we get the benefits for free without needing to remember to bring this back

Reply:
> But this is part of task fix which is not used in CI

True

> nilaway -fix isn't useful today but becomes useful later

I don't think it's a good idea for us to be autofixing potential `nil` pointer dereferences in any case. That would result in our code having excessive nil checks and mask the underlying design problem.

Reply:
Can you demonstrate this? `task fix` is an opt-in only "feature" during local development, is completely outside of automation and CI. If anything it gives us another opinion on our code that we can simply ignore. 
As an aside, this seems like a random change outside the scope of the added feature.

Reply:
@davidbloss Yes. This is an example of `nilaway` creating several alerts for the same problem repeatedly. 

```
$ cd repos/cli && git checkout main && git pull
$ nilaway -fix -test=false -include-pkgs='github.com/opslevel/cli' ./...
/Users/taimoorahmad/repos/cli/src/common/csv.go:26:9: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> common/csv.go:21:10: result 0 of `Read()` lacking guarding; assigned into field `Row`
	-> common/csv.go:26:9: field `Row` sliced into

/Users/taimoorahmad/repos/cli/src/common/helpers.go:39:14: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> common/helpers.go:39:14: read directly from variadic parameter `values` sliced into

/Users/taimoorahmad/repos/cli/src/common/prompts.go:74:10: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> common/prompts.go:74:10: unassigned variable `filteredList` sliced into

/Users/taimoorahmad/repos/cli/src/cmd/check.go:460:31: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> cmd/check.go:460:31: unassigned variable `input` accessed field `Version`

(Same nil source could also cause potential nil panic(s) at 1 other place(s): cmd/check.go:458:5.)

/Users/taimoorahmad/repos/cli/src/cmd/dependency.go:76:50: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> cmd/dependency.go:76:50: unassigned variable `in` accessed field `Source`

(Same nil source could also cause potential nil panic(s) at 2 other place(s): cmd/dependency.go:77:50, and cmd/dependency.go:79:25.)

/Users/taimoorahmad/repos/cli/src/cmd/property.go:197:11: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> cmd/property.go:197:11: unassigned variable `d` dereferenced

/Users/taimoorahmad/repos/cli/src/cmd/terraform.go:180:30: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> cmd/terraform.go:180:30: result 0 of `ListIntegrations()` lacking guarding; accessed field `Nodes`

/Users/taimoorahmad/repos/cli/src/cmd/terraform.go:191:11: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> cmd/terraform.go:191:11: result 0 of `ListRepositories()` lacking guarding; accessed field `Nodes`

/Users/taimoorahmad/repos/cli/src/cmd/terraform.go:284:23: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> cmd/terraform.go:284:23: result 0 of `GetTags()` lacking guarding; accessed field `Nodes`

/Users/taimoorahmad/repos/cli/src/cmd/terraform.go:273:14: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> cmd/terraform.go:273:14: result 0 of `ListServices()` lacking guarding; accessed field `Nodes`

/Users/taimoorahmad/repos/cli/src/cmd/terraform.go:332:11: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> cmd/terraform.go:332:11: result 0 of `ListTeams()` lacking guarding; accessed field `Nodes`

/Users/taimoorahmad/repos/cli/src/cmd/terraform.go:366:16: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> cmd/terraform.go:366:16: result 0 of `ListCategories()` lacking guarding; accessed field `Nodes`

/Users/taimoorahmad/repos/cli/src/cmd/terraform.go:403:25: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> cmd/terraform.go:403:25: result 0 of `ListFilters()` lacking guarding; accessed field `Nodes`

/Users/taimoorahmad/repos/cli/src/cmd/terraform.go:552:24: error: Potential nil panic detected. Observed nil flow from source to dereference point:
	-> cmd/terraform.go:552:24: result 0 of `ListChecks()` lacking guarding; accessed field `Nodes`


```

> If anything it gives us another opinion on our code that we can simply ignore.

`nilaway` without `-fix` is already a mandatory step on CI and `task lint`. If `-fix` worked it would encourage developers to just run it without thinking to silence CI alerts. 

> As an aside, this seems like a random change outside the scope of the added feature.

This is the equivalent of fixing a typo - it has no impact on an anything. I don't see how it is worth it to take the time to checkout this change to a new branch, type up a new PR description and then wait 1-24+ hours for review.

Reply:
A couple issue with this:
- This example is run against our [cli repo](https://github.com/opslevel/cli) and not this repo
   - It turns out `task lint` in the cli repo currently raises issues if `go.work` is not getting in the way.
- Running the nilaway checks in `opslevel-go` do not point anything out. This is why this change seems random to me.

> This is the equivalent of fixing a typo - it has no impact on an anything.

Fixing a typo has zero behavioral side effects. This change is different, as it changes `task fix` and therefore changes the local development workflow for anyone choosing to use `task fix`.

>  I don't see how it is worth it to take the time to checkout this change to a new branch, type up a new PR description and then wait 1-24+ hours for review.

I had no idea you felt like your PRs were not getting looked at quickly enough. More than half of my work week this week has been centered around PRs.
Please understand that the point of breaking out PRs is to quickly expedite individual concerns. Your [opslevel-jq-parser PR](https://github.com/OpsLevel/opslevel-jq-parser/pull/45) is a good example of a focused individual concern. Seemingly unrelated changes like this `task fix` change can slow down otherwise solid improvements to the main goal like adding `GetSystem()`

Reply:
>This example is run against our [cli repo](https://github.com/opslevel/cli) and not this repo

I cannot give you an example in this repo because `nilaway` has been rolled out for weeks and there are no alerts here currently. If you `git checkout` a v2023 release you should be able to see what nilaway evaluated before we reduced the amount of nil pointer situations. 

> Fixing a typo has zero behavioral side effects. This change is different

`nilaway -fix` does not make any changes whatsoever, checking for nil pointer dereferences is still 100% enabled by `task lint` and `task ci`. If `nilaway -fix` did work, we would not want to use it because the fixes it could suggest would be false positives/bandaid solutions like in the output related to `terraform.go`

> Seemingly unrelated changes like this task fix change can slow down otherwise solid improvements

I will remove this line.